caproto/threading/client.py

```diff
@@ -1789,6 +1792,110 @@ def __del__(self):
             pass
 
 
+class Batch:
+    """
+    Accumulate requests and then issue them all in batch.
+
+    Parameters
+    ----------
+    timeout : number or None
+        Number of seconds to wait before ignoring late responses. Default
+        is 2.
+
+    Examples
+    --------
+    Read some PVs in batch and stash the readings in a dictionary as they
+    come in.
+
+    >>> readings = {}
+
+    >>> results = {}
+    >>> stash_result = lambda name, response: results.update({name: response.data})
```

Comment:
flake8: no lambda! 😄 
```
def stash_result(name, response):
    results[name] = response.data
```

^ seems a bit more clear for a basic example, I think.